### PR TITLE
create TEST image for Pg17

### PR DESCRIPTION
**Pull request by thomasboussekey**

### CODE REVIEW

**Comment by phillipross** [comment]
@thomasboussekey thanks for your contribution!  Please ignore the coderabbit review for now, we're just testing it 😅

File: 17-beta/initdb-postgis.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+set -e
+
+# Perform all actions as $POSTGRES_USER
+export PGUSER="$POSTGRES_USER"
+
+# Create the 'template_postgis' template db
+"${psql[@]}" <<- 'EOSQL'
+CREATE DATABASE template_postgis IS_TEMPLATE true;
+EOSQL
```
[the remark this answers is not in the document]
Comment (Komzpa) [reply 17-beta/initdb-postgis.sh:11]: @coderabbitai remember, this is the way it should be, it gets set in the parent official Postgres docker image up the call stack.

**Comment by ImreSamu** [comment]
Hi @thomasboussekey , 

Thank you for the PR!

I have made some changes to the PR:
- Converted your PR to the `17beta2-master` directory
- Updated `.github/workflows/main.yml` and `README.md`
- Ran `./update.sh` to regenerate the Dockerfiles from the template

After reviewing the PR, I added other changes, such as the postgis `3.5.0alpha2` test images, and updated Alpine to 3.20.

So now we have three new images:

- `postgis/postgis:17beta2-master`
- `postgis/postgis:16-3.5.0alpha2-alpine`
- `postgis/postgis:17beta2-3.5.0alpha2-alpine`

@phillipross : 

- I believe I have added all necessary changes, and the PR is ready for review.
- The Alpine 3.18 to Alpine 3.20 upgrade is necessary due to upstream PostgreSQL Alpine changes (although Alpine 3.19 would also suffice).
- Previously, there were 12 image/jobs. With the new changes, there will be 15 image/jobs (+3), and these have successfully run.

**Review by ImreSamu — APPROVED** [review]

**Review by phillipross — APPROVED** [review]
looks great, thanks!

**Pull request merged**